velocity/src/main/java/me/leoko/advancedban/velocity/VelocityMethods.java

```diff
@@ -0,0 +1,377 @@
+import me.leoko.advancedban.velocity.command.CommandReceiverVelocity;
+import me.leoko.advancedban.velocity.event.PunishmentEvent;
+import me.leoko.advancedban.velocity.event.RevokePunishmentEvent;
+import me.leoko.advancedban.velocity.utils.LuckPermsOfflineUser;
+import net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer;
+import ninja.leaping.configurate.ConfigurationNode;
+import ninja.leaping.configurate.objectmapping.ObjectMappingException;
+import ninja.leaping.configurate.yaml.YAMLConfigurationLoader;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.slf4j.Logger;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+public class VelocityMethods extends AbstractMethodInterface<ConfigurationNode> {
+
+  private final ProxyServer server;
+  private final Path dataDirectory;
+  private final Logger logger;
+
+  private boolean luckPermsSupport;
+
+  public VelocityMethods(Path dataDirectory,
+                         final @NonNull ProxyServer server,
+                         Logger logger) {
+    super(dataDirectory);
+    this.dataDirectory = dataDirectory;
+    this.server = server;
+    this.logger = logger;
+
+    if (server.getPluginManager().getPlugin("luckperms").isPresent()) {
+      luckPermsSupport = true;
+      log("[AdvancedBan] Offline permission support through LuckPerms active");
+    } else {
+      luckPermsSupport = false;
+      log("[AdvancedBan] No offline permission support through LuckPerms");
+    }
+
+  }
+
+
+  @Override
+  protected ConfigurationNode loadConfiguration(Path configPath) throws IOException {
+      YAMLConfigurationLoader loader = YAMLConfigurationLoader.builder().setPath(configPath).build();
+      return loader.load();
+  }
+
+  @Override
+  public String getFromUrlJson(String url, String key) {
+    try {
+      HttpURLConnection request = (HttpURLConnection) new URL(url).openConnection();
+      request.connect();
+
+      JsonObject json = (JsonObject) JsonParser.parseReader(new InputStreamReader(request.getInputStream()));
+
+      String[] keys = key.split("\\|");
+      for (int i = 0; i < keys.length - 1; i++) {
+        json = json.getAsJsonObject(keys[i]);
+      }
+
+      return json.get(keys[keys.length - 1]).toString().replaceAll("\"", "");
+
+    } catch (Exception exc) {
+      return null;
+    }
+  }
+
+  @Override
+  public String getVersion() {
+    return "@version";
+  }
+
+  // Object.toString() is considered safe in this scenario (as per Configurate Docs)
+  @Override
+  public String[] getKeys(Object file, String path) {
+    return ((ConfigurationNode)file).getNode(path.split("\\.")).getChildrenMap().keySet().stream().map(Object::toString).toArray(String[]::new);
```

Comment:
Getting a node from a path is copied a lot. You could make a private method to reduce code duplication:
```java
private ConfigurationNode getConfigNode(Object file, String path) {
  return ((ConfigurationNode) file).getNode((Object[]) path.split("\\."));
}
```